File: lib/test/t_DAT_std.cxx
#include <iostream>
#include <cmath>

// OT includes
#include <openturns/OT.hxx>

#include "DAT.hxx"

using namespace OT;
using namespace OTPMML;

int main(int argc, char **argv)
{
  Description variables(4);
  Description formula(1);
  variables[0] = "E";
  variables[1] = "F";
  variables[2] = "L";
  variables[3] = "I";
  formula[0] = "F*L^3/(3*E*I)";
  SymbolicFunction poutre(variables, formula);

  Collection<Sample> samples(DAT::Import("input_output.dat"));
  std::cout << "Input:" << samples[0] << std::endl;
  std::cout << "Output:" << samples[1] << std::endl;
  const Sample result(poutre(samples[0]));
  std::cout << "Analytical output:" << result << std::endl;
  for (UnsignedInteger i = 0; i < result.getSize(); ++i)
  {
    if (std::abs(result[i][0] - samples[1][i][0]) > 1.e-10 * std::abs(result[i][0]))
    {
      std::cout << "Significant differences found at index " << i << std::endl;
      break;
    }
  }

  DAT::Export("export.dat", samples[0], samples[1]);

  return 0;
}

File: python/test/t_DAT_std.py
#! /usr/bin/env python

import openturns as ot
import otpmml

# Comparison of neuralNetwork with the original function
poutre = ot.SymbolicFunction(["E", "F", "L", "I"], ["F*L^3/(3*E*I)"])
print("poutre=", repr(poutre))

# Import data used for comparison
samples = otpmml.DAT.Import("input_output.dat")
input_sample = samples[0]
output_sample = samples[1]

print("input sample=", input_sample)
print("output sample=", output_sample)
result = poutre(input_sample)

print("Analytical output=", result)

relative_difference = [0] * result.getSize()
for i in range(result.getSize()):
    relative_difference.append(
        abs((result[i][0] - output_sample[i][0]) / result[i][0]))

if max(relative_difference) > 1.e-10:
    print("Relative difference=", relative_difference)

otpmml.DAT.Export("export.dat", input_sample, output_sample)
